File: python/vigenere.py
def caesar(key, message2, abc):
  size=len(message2)
  bool = False
  n = 0
  x = 0
  for i in range(size):
    bool = False
    n = 0
    while n < 26 and not bool:
      if message2[i] == abc[n]:
        s = n
        bool = True
      n = n + 1
    bool = False
    n = 0
    while n < 26 and not bool:
      if message2[i] != ' ':
        if x == len(key):
          x = 0
        if key[x] == abc[n]:
          t = n + s
          x = x + 1
          bool = True
      n = n + 1
    if message2[i] != ' ':
      if t >= 26:
        r = t - 26
        message2[i] = abc[r]
      else:
        message2[i] = abc[t]

message = "usa software libre"
message2 = list(message)
key = "casa"
print("Mensaje original: ", message)
abc=['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
caesar(key, message2, abc)
print("Mensaje encriptado: ", "".join(message2))

File: c/vigenere.c
#include <stdio.h>
#include <string.h>

int vigenere(char message[], char abc[], char key[]);
void main(void)
 {
  setvbuf(stdout, NULL, _IONBF, 0);
  
  char message[] = "usa software libre";
  char key[] = "casa";
  
  char abc[26] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
  
  printf("Mensaje original: %s", message);
  vigenere(message, abc, key);
  printf("\nMensaje cifrado: %s", message);
 }
int vigenere(char message[], char abc[], char key[])
 {
  int b = 0, i, n = 0, r, s, t, x = 0;
  for(i = 0; i < strlen(message); i++)
   {
    b = 0;
    n = 0;
    while(n < 26 && !b)
     {
      if(message[i] == abc[n])
       {
        s = n;
        b = 1;
       }
      n = n + 1;
     }
    b = 0;
    n = 0;
    while(n < 26 && !b)
     {
      if(message[i] != ' ')
       {
       	if(x == strlen(key))
       	 {
       	  x = 0;
       	 }
        if(key[x] == abc[n])
         {
          t = n + s;
          x = x + 1;
          b = 1;
         }
       }
      n = n + 1;
     }
    if(message[i] != ' ')
     {
      if(t >= 26)
       {
        r = t - 26;
        message[i] = abc[r];
       }
      else
       {
       	message[i] = abc[t];
       }
     }
   }
 }
